src/ObjectPromiseProxy.js

```diff
@@ -26,9 +26,8 @@ function ObjectPromiseProxy (promise, target) {
             target.store.createModelsFromData(json.included)
           }
         })
-        // Update target isInFlight and isDirty
+        // Update target isInFlight
         target.isInFlight = false
-        target.isDirty = false
```

Comment:
When the object has been synced with the server I don't think it should be considered dirty anymore  right?

Reply:
it will be clean because we will take a new snapshot which will "clear" the dirty state